### PR TITLE
Authentication with devise

### PR DESCRIPTION
**Pull request by ashraffares**
building a simple blog-post app that has user authentication with devise gem
- Add making posts functionality 
- add devise gem
- authenticate user  and allow certain functionally if is it not
- only show post author if the user authenticated with before action
- add is it correct_user method to validate the user and add it to before action

### CODE REVIEW

**Review by umairarshadbutt — CHANGES_REQUESTED** [review]
# Status: INVALID :red_circle: 

**Hi @ashraffares **
- [x] The project has no descriptive README.md. plrease add the descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
I am going to mark this project as invalid so when you fix the issue just submit another code review request.
##

 **If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)
t

**Comment by ashraffares** [comment]
@umairarshadbutt  
Add readme file done

**Comment by umairarshadbutt** [comment]
Hi @ashraffares 
Please submit it again for review

File: README.md [hunk 1]
```diff
@@ -1,24 +1,57 @@
...
 
-* How to run the test suite
+## Authors
 
-* Services (job queues, cache servers, search engines, etc.)
+👤 **Fares Ashraf** 
 
-* Deployment instructions
+- twitter:https://twitter.com/Fares09301164
+- GitHub: https://github.com/ashraffares/http-ashraffares.github.io-
```
Comment (descholar-ceo) [inline README.md]: [required] Your github profile link is broken, kindly fix it

File: README.md [hunk 2]
```diff
@@ -1,24 +1,57 @@
...
+## Getting Started
+
+**This is an example of how you may give instructions on setting up your project locally.**
+**Modify this file to match your project, remove sections that don't apply. For example: delete the testing section if the currect project doesn't require testing.**
+
+
+To get a local copy up and running follow these simple example steps.
+- Clone this repository with: git clone ```https://github.com/ashraffares/devise-app.git``` using your terminal or command line.
+- Change to the project directory by entering : <br>
+```cd members-only``` in the terminal .
```
Comment (descholar-ceo) [inline README.md]: [required] This command is a bit erroneous, the directory to enter is not `members-only` it should be `devise-app` instead

File: README.md [hunk 3]
```diff
@@ -1,24 +1,57 @@
...
+
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](issues/).
```
Comment (descholar-ceo) [inline README.md]: [required] This issues link is broken, please replace this one with the appropriate one

File: README.md [hunk 4]
```diff
@@ -1,24 +1,57 @@
...
+- twitter:https://twitter.com/Fares09301164
+- GitHub: https://github.com/ashraffares/http-ashraffares.github.io-
+- linkedin:https://www.linkedin.com/in/fares-ashraf-382a35176/
 
-* ...
+## Getting Started
+
+**This is an example of how you may give instructions on setting up your project locally.**
+**Modify this file to match your project, remove sections that don't apply. For example: delete the testing section if the currect project doesn't require testing.**
+
```
Comment (descholar-ceo) [inline README.md]: [required] You need to remove this text, it's no longer needed here, the purpose of this was to tell you what to do

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    @posts = Post.all
+    @user = User.all
+  end
+
+  # GET /posts/1 or /posts/1.json
+  def show; end
+
+  # GET /posts/new
+  def new
+    # @post = Post.new
```
Comment (descholar-ceo) [inline app/controllers/posts_controller.rb]: [required] Let's remove all of the commented codes

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @ashraffares  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implement all of the minimal requirements of this project, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] In order to make your repo more professional, read and fix the inline comments I put in your readme file
- [ ] There are some commented codes in your project, please remove them out
- [ ] You have some logic in views: View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)
  - [ ] There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright _=> To make sure that I clarify what you need to change on this reqirement: Ensure that there is no if conditions, and no loops in your view (`.html.erb`) files_
  
#### Resources to help you learn about this:

- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code/)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

File: app/controllers/posts_controller.rb (hunk 5 above)
Comment (ashraffares) [reply app/controllers/posts_controller.rb]: done

File: README.md (hunk 4 above)
Comment (ashraffares) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (ashraffares) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (ashraffares) [reply README.md]: done

**Review by umairarshadbutt — APPROVED** [review]
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @ashraffares ! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
**If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Pull request merged**